### PR TITLE
Feedback collection

### PR DESCRIPTION
**Pull request by nghtctrl**
Review request for Souad's feedback collection app.

### CODE REVIEW

**Comment by nghtctrl** [comment]
@souadyakubu it would be nice to store the codes into a directory within this repository (i.e. a directory named `feedback-collection`) instead of pushing it directly into the parent directory. It would help us be more organized, and it would be easier to keep your code in one place.

File: feedback_tool.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
+import psycopg2
+import random
+import streamlit as st
+from transformers import BartForConditionalGeneration, BartTokenizer
+
+# establishing connection to the database used
+my_connection = psycopg2.connect(database="postgres", user="souadyakubu", password="souadseny", host="localhost", port="5432")
```
Comment (kcarnold) [inline feedback_tool.py:7]: We definitely don't want secrets in our git repo. Alternative: https://docs.streamlit.io/library/advanced-features/secrets-management

File: feedback_tool.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+cursor = my_connection.cursor()
```
Comment (kcarnold) [inline feedback_tool.py:8]: I'm unsure it's a good idea to have a cursor globally. I think you can do `with connection: connection.execute(...)` see https://docs.python.org/3/library/sqlite3.html#sqlite3-connection-context-manager

File: feedback_tool.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    selected_question_index = st.selectbox("Select the best question:", [i + 1 for i in range(len(random_questions))])
+
+    rank = st.slider("Rate the question (1 = worst, 5 = best):", 1, 5)
+
+    if st.button("Submit Feedback"):
+        selected_question = random_questions[selected_question_index - 1]
+        system_that_generated = "hyechanjun/interview-question-remake"
+        rater_id = 1  #can be changed based on user authentication
+        feedback_log(random_context, selected_question, system_that_generated, rater_id, rank)
+        st.success("Feedback submitted successfully!")
```
Comment (kcarnold) [inline feedback_tool.py:68]: What's the flow for collecting more than one feedback?

File: feedback_tool.py [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+if __name__ == "__main__":
```
Comment (kcarnold) [inline feedback_tool.py:70]: This idiom isn't needed for streamlit apps; I'd just put the `main` stuff globally.

File: feedback_tool.py [hunk 5]
```diff
@@ -0,0 +1,71 @@
+import psycopg2
+import random
+import streamlit as st
+from transformers import BartForConditionalGeneration, BartTokenizer
```
Comment (kcarnold) [inline feedback_tool.py:4]: I think we should have some other module or even other process that's generating the contexts and questions; this one can just handle collecting the ratings.

**Review by kcarnold — APPROVED** [review]
Looking good to me so far, with the only big deal being having a secret included. For now, just assume that password is known to the world and use a new password going forward.

**Review by souadyakubu — APPROVED** [review]

**Pull request merged**